packages/plugin-source-build/src/plugin.ts

```diff
@@ -58,15 +58,17 @@ export function pluginSourceBuild(
     setup(api) {
       const projectRootPath = api.context.rootPath;
 
-      let projects: Project[] = [];
-
-      api.modifyRsbuildConfig(async (config) => {
-        projects = await getDependentProjects(projectName || projectRootPath, {
-          cwd: projectRootPath,
-          recursive: true,
-          filter: filterByField(sourceField, true),
-          extraMonorepoStrategies,
-        });
+      let projects: Project[] | undefined;
+
+      api.modifyEnvironmentConfig(async (config) => {
+        projects =
+          projects ||
```

Comment:
What if each environments use the different projects name?

If we want to cache `project` in the global, we may need to create a `Map` and use `projectName || projectRootPath` as the cache key.

Reply:
This is just to ensure the performance between different Environments of the global plugin. `different projects name` will only appear when plugins are registered separately.

Reply:
Ok I get it